File: ambeth/Ambeth.Persistence.API/ambeth/persistence/ILink.cs
using System;
using System.Collections;

namespace De.Osthus.Ambeth.Persistence
{
    public interface ILink
    {
        ILinkMetaData MetaData { get; }

        ITable FromTable { get; }

        ITable ToTable { get; }
        
        IDirectedLink DirectedLink { get; }

        IDirectedLink ReverseDirectedLink { get; }
        
        ILinkCursor FindAllLinked(IDirectedLink fromLink, IList fromIds);

        ILinkCursor FindAllLinkedTo(IDirectedLink fromLink, IList toIds);

        ILinkCursor FindLinked(IDirectedLink fromLink, Object fromId);

        ILinkCursor FindLinkedTo(IDirectedLink fromLink, Object toId);

        void LinkIds(IDirectedLink fromLink, Object fromId, IList toIds);

        void UpdateLink(IDirectedLink fromLink, Object fromId, Object toIds);

        void UnlinkIds(IDirectedLink fromLink, Object fromId, IList toIds);

        void UnlinkAllIds(IDirectedLink fromLink, Object fromId);

        void UnlinkAllIds();

        void StartBatch();

        int[] FinishBatch();

        void ClearBatch();
    }
}

File: jambeth/jambeth-persistence-api/src/main/java/com/koch/ambeth/persistence/api/ILink.java
package com.koch.ambeth.persistence.api;

import java.util.List;

public interface ILink
{
	ILinkMetaData getMetaData();

	ITable getFromTable();

	ITable getToTable();

	IDirectedLink getDirectedLink();

	IDirectedLink getReverseDirectedLink();

	ILinkCursor findAllLinked(IDirectedLink fromLink, List<?> fromIds);

	ILinkCursor findAllLinkedTo(IDirectedLink fromLink, List<?> toIds);

	ILinkCursor findLinked(IDirectedLink fromLink, Object fromId);

	ILinkCursor findLinkedTo(IDirectedLink fromLink, Object toId);

	void linkIds(IDirectedLink fromLink, Object fromId, List<?> toIds);

	void updateLink(IDirectedLink fromLink, Object fromId, Object toIds);

	void unlinkIds(IDirectedLink fromLink, Object fromId, List<?> toIds);

	void unlinkAllIds(IDirectedLink fromLink, Object fromId);

	void unlinkAllIds();

	void startBatch();

	int[] finishBatch();

	void clearBatch();
}
